pkg/cloudlogging/puller.go

```diff
@@ -111,3 +113,47 @@ func (p *Puller) Pull(ctx context.Context, filter string, maxCount int) ([]*logg
 	}
 	return ls, nil
 }
+
+// SteamPull pulls live log entries, it stops pulling after logNum of log entries.
+func (p *Puller) SteamPull(ctx context.Context, filter string, logNum int, result chan<- []*loggingpb.LogEntry) error {
+	logger := logger.FromContext(ctx)
+	if err := retry.Do(ctx, p.retry, func(ctx context.Context) error {
+		stream, err := p.client.TailLogEntries(ctx)
+		if err != nil {
+			return retry.RetryableError(fmt.Errorf("failed to create stream: %w", err))
+		}
+		defer func() {
+			if err := stream.CloseSend(); err != nil {
+				logger.Infof("failed to close stream: %w", err)
+			}
+		}()
+		req := &loggingpb.TailLogEntriesRequest{
+			ResourceNames: []string{p.resource},
+			Filter:        filter,
+		}
+		if err := stream.Send(req); err != nil {
+			return retry.RetryableError(fmt.Errorf("failed to send request: %w", err))
+		}
+
+		for counter := 0; counter < logNum; {
```

Comment:
Forgot to mention one of the feedback I got from Seth and Chen is to simulate the behaviour of linux [tail -f ](https://www.geeksforgeeks.org/tail-command-linux-examples/), let's get rid of logNum, and abort when interrupted.

Reply:
Remove logNum